src/routeGeneration/routeGenerator.ts

```diff
@@ -31,7 +32,11 @@ export class RouteGenerator {
   constructor(private readonly metadata: Tsoa.Metadata, private readonly options: RoutesConfig, private readonly minimalSwaggerConfig: SwaggerConfigRelatedToRoutes) {}
 
   public async GenerateRoutes(middlewareTemplate: string, pathTransformer: (path: string) => string) {
-    const fileName = `${this.options.routesDir}/routes.ts`;
+    if (!fs.lstatSync(this.options.routesDir).isDirectory()) {
+      throw new Error(`routesDir should be a directory`);
+    }
+
+    const fileName = `${this.options.routesDir}/${this.options.routesFileName || 'routes.ts'}`;
```

Comment:
I should have noticed this before but we need to check that routesFileName ends in .ts. Not if you use .endsWith please first check if the version of NodeJS that we set in package.json’s engine section supports it.

Reply:
I'm not sure why such a check would be necessary, I agree that helpful "warnings" (such as the directory check) can be useful, but this seems like unnecessary hand-holding to me. Maybe I do actually want my generated file to end in something else than `.ts` 🤷‍♂ (can't think of a reason why right now but it's a possibility).
And if someone actually forgot the `.ts` extension it's a 5 second fix, so it doesn't *really* hurt DX in my opinion.
Unlesss there's an actual technical reason why it could cause issues of course, but I can't think of anything non-IDE related that would unconditionally break with a "wrong" file extension.

If you really want it though I'll be happy to add it.

> if you use .endsWith please first check if the version of NodeJS that we set in package.json’s engine section supports it.

An automated test environment could check for these sort of things by running the tests on multiple engines.

Reply:
I'm sorry for my broken early-morning English, what I'm trying to say is: I feel like this (in my opinion) unnecessary hand-holding reduces flexibility (albeit for a small group) while not really improving DX. In fact it could be hurting DX for those who want their file to end in something other than `.ts`.

Reply:
@HoldYourWaffle my primary concern as a volunteer who works on tsoa is to provide functionality that helps the majority of users and that prevents future issues coming in. If even 1 person writes an issue about “hey, why didn’t routesFileName work for me” then it has wasted my free time. Additionally, any erroneous issue created on tsoa’s github makes tsoa look like a less professional repository. So my goal is at all times to diminish the number of times that someone has to reach out to tsoa. The library should provide the support itself.

We need to prioritize the goals and features of the majority user over the fringe case users.

And to be 100% clear, the routes.ts file is in fact a TypeScript file. So if you don’t give it a .ts extension then the person’s project won’t build. Tsoa can not be responsible for that.

Reply:
> If even 1 person writes an issue about “hey, why didn’t routesFileName work for me” then it has wasted my free time

That's an interesting viewpoint. I wholeheartedly agree that "sanity checks" are useful, this just seems a little overkill in my opinion. It's really not that difficult to figure out why a file doesn't get an extension if you didn't tell it to.
I have added the check now.

> We need to prioritize the goals and features of the majority user over the fringe case users.

Don't get me wrong, I definitely agree on this. I just think that it's important to *allow* "edge-case" configurations if it's reasonably possible to do so; mostly because I've had to fork multiple projects that didn't want to cooperate in the past (the origin of this PR) and that's just annoying. This isn't the fault of the library per se, but it's always nice to avoid such things.

> And to be 100% clear, the routes.ts file is in fact a TypeScript file. So if you don’t give it a .ts extension then the person’s project won’t build. Tsoa can not be responsible for that.

It wouldn't be TSOA's responsibility, because the user is the one that forgot the extension.

Writing this, maybe a warning would be better suited in this situation? Users that want to do non-standard stuff can still do so but the forgetful ones still get a nice little heads-up. If you want me to leave it as-is that's of course fine too, you're the maintainer after all 😉